src/Lecture3.hs

```diff
@@ -122,9 +148,11 @@ data Reward = Reward
     } deriving (Show, Eq)
 
 instance Semigroup Reward where
-
-
+--  a <> b = Reward (rewardGold a <> rewardGold b) (rewardSpecial a || rewardSpecial b)
+  Reward a b <> Reward c d = Reward (a <> c) (b || d)
+  
 instance Monoid Reward where
+  mempty = Reward (Gold 0) False
```

Comment:
You can use `mempty` for `Gold` here because it already has a `Monoid` instance 🏅 
```suggestion
  mempty = Reward mempty False
```